src/commands/init.ts

```diff
@@ -52,183 +53,128 @@ export default class Init extends Command {
     help: flags.help({
       char: 'h',
       description: 'Show help for the new command.'
+    }),
+    verbose: flags.boolean({
+      char: 'v',
+      description: 'Show more detailed output.'
     })
   };
 
-  async run() {
-    const { flags } = this.parse(Init);
+  static args = [
+    {
+      name: 'projectDir',
+      description: 'Path to the folder in which the project should be created.'
+    }
+  ];
 
-    ensureDirectoryExistence(flags.projectDir, true);
+  async run() {
+    const { flags, args } = this.parse(Init);
+    const { verbose } = flags;
 
-    const initType = await this.determineInitializationType(flags);
-    await this.initProject(flags, initType);
+    if (typeof flags.projectDir !== 'undefined' && typeof args.projectDir !== 'undefined' && flags.projectDir !== args.projectDir) {
+      this.error('Project directory was given via argument and via the `--projectDir` flag. Please only provide one.', { exit: 1 });
```

Comment:
We could add the provided values for the mismatching flag-arg pair to better show the mismatch to the user here.